src/test/java/picard/illumina/parser/TileMetricsUtilTest.java

```diff
@@ -2,73 +2,93 @@
 
 import htsjdk.samtools.util.IOUtil;
 import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.*;
 import java.util.stream.IntStream;
 
 public class TileMetricsUtilTest {
-    @Test
-    public void testFindTileMetrics() throws IOException {
-        Path baseDir = createTestDirs(0, true);
-        File tileMetricsFile = TileMetricsUtil.renderTileMetricsFileFromBasecallingDirectory(
-                baseDir.toFile(), 0, false);
 
-        Assert.assertEquals(tileMetricsFile.getAbsolutePath(),
-                baseDir.resolve(TileMetricsUtil.INTEROP_SUBDIRECTORY_NAME)
-                        .resolve(TileMetricsUtil.TILE_METRICS_OUT_FILE_NAME).toString());
+    private Path testDir;
+
+    @BeforeClass
+    public void setupTestDir() throws IOException {
+        this.testDir = Files.createTempDirectory("TileMetricsUtilTests");
+    }
 
-        IOUtil.deleteDirectoryTree(baseDir.toFile());
+    @AfterClass
+    public void cleanupTestDir() {
+        IOUtil.deleteDirectoryTree(testDir.toFile());
+    }
+
+    @DataProvider
+    public Object[][] testCases() {
+        List<Integer> oneCycleFile = Collections.singletonList(1);
+        List<Integer> allCycleFiles = Arrays.asList(1, 2, 3, 4, 5);
+        List<Integer> someCycleFiles = Arrays.asList(2, 4);
+
+        return new Object[][]{
+            //specificTestDir, numCycleMetricsFiles, populatedCycleDirs, baseMetricsFile, isNovaSeq
+            {"notNovaSeq", 0, Collections.emptyList(), true, false},
+            {"novaSeqBaseFileOnly", 0, Collections.emptyList(), true, true},
+            {"novaSeqSingleCycleFile", 1, oneCycleFile, false, true},
+            {"novaSeqBaseAndSingleCycleFile", 1, oneCycleFile, true, true},
+            {"novaSeqAllCycleFiles", 5, allCycleFiles, false, true},
+            {"novaSeqBaseAndAllCycleFiles", 5, allCycleFiles, true, true},
+            {"novaSeqSomeCycleFiles", 5, someCycleFiles, false, true},
+            {"novaSeqBaseAndSomeCycleFiles", 5, someCycleFiles, true, true}
```

Comment:
What is going on here?

Reply:
These are test inputs that are run through testFilePresencePermutation. Whether the base directory is populated or not, and whether all, none, one or some non-sequential cycle directories contain TileMetricsOut.bin files, findTileMetricsFiles should identify and return a list of all of them